x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -2003,6 +2006,31 @@ private void reloadRemoteClusterCredentials(Settings settingsWithKeystore) {
         future.actionGet();
     }
 
+    public Map<String, String> getAuthContextForSlowLog() {
+        if (this.securityContext.get() != null && this.securityContext.get().getAuthentication() != null) {
+            Authentication authentication = this.securityContext.get().getAuthentication();
+            Subject authenticatingSubject = authentication.getAuthenticatingSubject();
+            Subject effetctiveSubject = authentication.getEffectiveSubject();
+            Map<String, String> authContext = new HashMap<>();
+            if (authenticatingSubject.getUser() != null) {
+                authContext.put("user.name", authenticatingSubject.getUser().principal());
+                authContext.put("user.realm", authenticatingSubject.getRealm().getName());
+            }
+            // Only include effective user if different from authenticating user (run-as)
+            if (effetctiveSubject.getUser() != null && effetctiveSubject.equals(authenticatingSubject) == false) {
+                authContext.put("user.effective.name", effetctiveSubject.getUser().principal());
+                authContext.put("user.effective.realm", effetctiveSubject.getRealm().getName());
+            }
+            authContext.put("auth.type", authentication.getAuthenticationType().name());
+            if (authentication.isApiKey()) {
+                authContext.put("apikey.id", authenticatingSubject.getMetadata().get(AuthenticationField.API_KEY_ID_KEY).toString());
+                authContext.put("apikey.name", authenticatingSubject.getMetadata().get(AuthenticationField.API_KEY_NAME_KEY).toString());
```

Comment:
I think it would be nice to include the `apikey.principal`=`authenticatingSubject.getUser().principal()` which is API key creator's user principal.  This is most like the easiest way to tie the request back to a user. 

Also, if we add `user.full_name` (comment above) we should (conditionally) add it here for the same (SAML) reason.

Reply:
`authenticatingSubject.getUser().principal()` is already available in `user.name`, do you mean we should duplicate it (it's always set if available)?

Reply:
doh ! you are right. disregard this comment. (mistakenly thought the api was an if/else w.r.t. user)